### PR TITLE
feat(escalating-v2): Emit escalating metrics 

### PR DESCRIPTION
**Pull request by snigdhas**
These changes from #52774 were reverting - undoing the revert.  

Update the ESCALATING_ISSUES use case during post_process as we see new events come in.

### CODE REVIEW

**Comment by snigdhas** [comment]
There's some relay tests failing because of the call to the metrics backend - @onewland @ayirr7 any chance you know how to fix these?

**Comment by ayirr7** [comment]
@snigdhas let me look into these

**Comment by ayirr7** [comment]
One note in terms of the rollout of this feature, can we put it behind a feature flag?

**Comment by snigdhas** [comment]
@ayirr7 do you mean the changes in this PR? Or the followup PR with the query? #53177

**Comment by snigdhas** [comment]
@getsentry/ingest could one of y'all help figure out the tests that are failing here? I'm not sure it's specifically related to the changes in this PR - I think the tests may need to be changed a bit to support the generic metrics pipeline.

**Comment by ayirr7** [comment]
> @ayirr7 do you mean the changes in this PR? Or the followup PR with the query? #53177

I think for this PR, so that we can easily turn on/off metrics emission as needed.

**Comment by ayirr7** [comment]
@getsentry/owners-ingest The tests are not failing now because the feature flag is not enabled, but can we have some help with the test failures? specifically https://github.com/getsentry/sentry/actions/runs/5798927319/job/15717780956?

**Comment by ayirr7** [comment]
@snigdhas I think I might have managed to fix those tests. I think it might still be good to have someone from the relevant team to take a look and give this their approval

**Comment by ayirr7** [comment]
Seems like tests are passing now. I think it's a good idea to have any relevant team(s) -- Issues, Ingest -- take a look at the way the tests have been changed and get approval from them as well before merging.

**Review by Swatinem — APPROVED** [review]

**Comment by ayirr7** [comment]
(Just fyi I will need to merge https://github.com/getsentry/getsentry/pull/11387 before this can be merged)

**Review by NisanthanNanthakumar — APPROVED** [review]

File: tests/sentry/notifications/test_notifications.py
Comment (ayirr7) [inline tests/sentry/notifications/test_notifications.py:464]: Do we know if the `response_num` needs to be passed in explicitly like this still? This was a different attempt to fix the tests.

File: src/sentry/conf/server.py
Comment (snigdhas) [reply src/sentry/conf/server.py:1382]: good point - reverted

Comment (ayirr7) [reply src/sentry/conf/server.py:1382]: Shall we make this flag set to `True` on this line so we could make sure tests are passing?

Comment (snigdhas) [reply src/sentry/conf/server.py:1382]: The only change is in `test_notifications` - it passes with the flag locally and the `get_attachment` helper isn't used elsewhere so we should be alright. Don't want to wait for 2 runs of CI 😅

Comment (ayirr7) [reply src/sentry/conf/server.py:1382]: Ok, makes sense

**Review by ayirr7 — APPROVED** [review]

**Pull request merged**
